installer/datafiles/Base_SCXCore.data

```diff
@@ -260,6 +260,12 @@ GenerateCertificate() {
 	    rm -f /etc/opt/omi/ssl/.omi_cert_marker
 	    rm -f ${{OMI_SSL_DIR}}/omikey.pem_temp ${{OMI_SSL_DIR}}/omi.pem_temp
         fi
+
+        if [ -f ${{OMI_SSL_DIR}}/omikey.pem ]; then
+            user_exists=$(id -u omi > /dev/null 2>&1; echo $?)
+            group_exists=$(id -g omi > /dev/null 2>&1; echo $?)
+            [ $user_exists = 0 -a $group_exists = 0 ] && chown omi:omi ${{OMI_SSL_DIR}}/omikey.pem
+        fi
```

Comment:
The problem with putting this in postinstall, which I mentioned in the E-mail about this issue, is that customers (or discovery) can run `scxsslconfig` after installation. If either of these happen, then the permissions would be messed up again, wouldn't they?

With your fix, what happens if you manually run `scxsslconfig -f` after installation? What's the permissions of this file?

Reply:
I have verified that if omikey.pem file exist by running scxsslconfig command ownership not getting modified.

Reply:
Change in ownership, not permission, of the file restricting omiengine to read omikey.pem file. 

As the owner of this file changed to root:root and permission with file is "S_IRUSR", which means read by owner, hence any program not running with root permission cannot read this file. Due to this omiengine which runs as omi:omi cannot read this.

With the new changes now ownership is omi:omi, permission is "S_IRUSR. Hence any program running with omi permission can read this file. 

As mentioned  "scxsslconfig -f" does not affect ownership. It only changes permission to read only and which is appropriate as it is private key file.
  
Let me know in case you have further concerns.